src/equisolve/numpy/_selection.py

```diff
@@ -0,0 +1,110 @@
+# SPDX-License-Identifier: BSD-3-Clause
+from typing import List, Type
+
+import skmatter._selection
+from equistore import TensorMap
+from equistore.operations import slice_block
+
+
+class GreedySelector:
+    """Wraps :py:class:`skmatter._selection.GreedySelector` for a TensorMap.
+
+    The class creates a selector for each block. The selection will be done based
+    the values of each :py:class:`TensorBlock`. Gradients will not be considered for
+    the selection.
+    """
+
+    def __init__(
+        self,
+        selector_class: Type[skmatter._selection.GreedySelector],
+        selection_type: str,
+        **selector_arguments,
+    ) -> None:
+        self._selector_class = selector_class
+        self._selection_type = selection_type
+        self._selector_arguments = selector_arguments
+
+        self._selector_arguments["selection_type"] = self._selection_type
+        self._selectors = {}
+
+    @property
+    def selector_class(self) -> Type[skmatter._selection.GreedySelector]:
+        """The class to perform the selection."""
+        return self._selector_class
+
+    @property
+    def selection_type(self) -> str:
+        """Whether to choose a subset of columns ('feature') or rows ('sample')."""
+        return self._selection_type
+
+    @property
+    def selector_arguments(self) -> dict:
+        """Arguments passed to the ``selector_class``."""
+        return self._selector_arguments
+
+    @property
+    def selectors(self) -> List[Type[skmatter._selection.GreedySelector]]:
+        """List of selector instances for each block."""
+        return self._selectors
+
+    def fit(self, X: TensorMap, warm_start: bool = False) -> None:
+        """Learn the features to select.
+
+        :param X:
+            Training vectors.
+        :param warm_start:
+            Whether the fit should continue after having already run, after increasing
+            `n_to_select`. Assumes it is called with the same X.
+        """
+        if len(X.components_names) != 0:
+            raise ValueError("Only blocks with no components are supported.")
+
+        for key, block in X:
+            selector = self.selector_class(**self.selector_arguments)
+            self._selectors[key] = selector.fit(block.values, warm_start=warm_start)
+
+        return self
+
+    def transform(self, X: TensorMap) -> TensorMap:
+        """Reduce X to the selected features.
+
+        :param X:
+            The input tensor.
+        :returns:
+            The selected subset of the input.
+        """
+        if len(self.selectors) == 0:
+            raise ValueError("No selections. Call fit method first.")
+
+        blocks = []
+        for key, block in X:
+            try:
+                selector = self.selectors[key]
+            except KeyError as err:
+                raise ValueError(f"Block with key {key} does not exist.") from err
+
+            mask = selector.get_support()
```

Comment:
I think we need to store the mask = selector.get_support() into a tensorblock (or the selected idx) somewhere in the fit function to be compatible with torch.nn.Module. I am not sure how you to do this operation with equistore only but maybe you have an idea?
```python
blocks.append(slice_block(block, properties=block.properties[mask]))
```

Reply:
Sounds related to https://github.com/lab-cosmo/equistore/issues/187

Reply:
> I think we need to store the mask = selector.get_support() into a tensorblock (or the selected idx) somewhere in the fit function to be compatible with torch.nn.Module. I am not sure how you to do this operation with equistore only but maybe you have an idea?
> 
> ```python
> blocks.append(slice_block(block, properties=block.properties[mask]))
> ```

After discussing with @agoscinski  we agreed that we only store the support as List[List[int]]. This allows the direct usage of torch later.





> Sounds related to [lab-cosmo/equistore#187](https://github.com/lab-cosmo/equistore/issues/187)

Yes, a bit. If we store a list of ints we can change the logic of slice block later once we have this functionality.